azure-pipelines.yml

```diff
@@ -23,12 +23,28 @@ variables:
   buildNumber: $[ counter('build', 400) ] # Start higher than our AppVeyor versions. Every build (pr or branch) will increment.
 
 steps:
+- pwsh: |
+    $releaseBranches = @('v4.x/ps7.2', 'v4.x/ps7.0', 'v3.x/ps7', 'v3.x/ps6', 'v2.x')
+    $isReleaseBuild = ($releaseBranches -contains $env:BuildSourceBranchName)
+    Write-Host "##vso[task.setvariable variable=IsReleaseBuild]$isReleaseBuild"
+    Write-Host "IsReleaseBuild: $isReleaseBuild"
+  displayName: 'Set IsReleaseBuild variable'
+  env:
+    BuildSourceBranchName: $(Build.SourceBranchName)
+
 - pwsh: ./build.ps1 -NoBuild -Bootstrap
   displayName: 'Running ./build.ps1 -NoBuild -Bootstrap'
 
 - pwsh: |
       $ErrorActionPreference = "Stop"
-      ./build.ps1 -Clean -Configuration Release -BuildNumber "$(buildNumber)" -AddSBOM -SBOMUtilSASUrl $env:SBOMUtilSASUrl
+      if ($isReleaseBuild)
+      {
+        ./build.ps1 -Clean -Configuration Release -BuildNumber "$(buildNumber)" -AddSBOM -SBOMUtilSASUrl $env:SBOMUtilSASUrl
+      }
+      else
+      {
+        ./build.ps1 -Clean -Configuration Release -BuildNumber "$(buildNumber)"
+      }
```

Comment:
non-blocking nit: Given the fact that the command we're running here shares the same prefix in both branches, I wonder if you could dynamically construct this command instead.

For example, I'm thinking of something like in the following **pseudocode**.
```
command <- '"./build.ps1 -Clean -Configuration Release -BuildNumber "$(buildNumber)"'
if ($isReleaseBuild)
{
  // add SBOM step
  command <- command  + "-AddSBOM -SBOMUtilSASUrl $env:SBOMUtilSASUrl"
}

Invoke(command);
```

That said, I realize this is just a configuration file so perhaps this is an unnecessary refactoring, so I'm leaving this as a non-blocking suggestion :)

Reply:
Thanks @davidmrdavid this is a good point. However, I think calling the script directly with the required parameters for each scenario is easier to read and maintain. For example, to debug a scenario where we build a command, we might have to add additional log statements to see what command is ran. If it is ok with you, I will keep it as is.

Reply:
Seems reasonable, merge ahead!